### PR TITLE
Gemini criware vs2022

### PR DESCRIPTION
**Pull request by Gemini-Loboto3**
Criware module replacement.

### CODE REVIEW

File: Include/criware/criware_server.cpp [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+		LeaveCriticalSection(&ADX_crit);
+}
+
+DWORD WINAPI server_thread(LPVOID params)
+{
+	while (loop)
+	{
+		adxs_Update();
+		Sleep(1);	// just give the thread enough time for locking from external operations
+					// used to be 10 milliseconds
```
Comment (elishacloud) [inline Include/criware/criware_server.cpp:48]: Why not use `Sleep(0);` ?  This way it returns to the thread as soon as other threads are done.

See:  https://stackoverflow.com/questions/3727420/significance-of-sleep0

**Comment by elishacloud** [comment]
@Gemini-Loboto3, I pushed some changes to your branch.  You may want to pull the latest changes from your branch before you continue working on the threading update.

**Comment by Polymega** [comment]
Thanks, Elisha! This PR should already cover 1.1 and DC as well, yes? If so, once you have the merge fully done, may I ask you to include a new test build here or on the main GitHub ticket? I am planning on making a public call-out to test this afterwards. 

This build should also have `SingleCoreAffinity` removed and `SingleCoreAffinityLegacy` not shown in the INI and disabled by default, correct?

**Comment by elishacloud** [comment]
Correct.  More details can be seen [here](https://github.com/elishacloud/Silent-Hill-2-Enhancements/issues/423#issuecomment-1110445683).

**Pull request merged**